### PR TITLE
Fixes thank you letter

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/2718

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- enable rewards
- go to regular site and tip that site
- make sure that thank you banner shows first letter of that site domain
- go to youtube video
- tip that creator and make sure that first letter of the channel name is displayed in thank you banner

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat_publishers.cc [hunk 1]
```diff
@@ -903,6 +903,8 @@ void BatPublishers::setBalanceReportItem(ledger::ACTIVITY_MONTH month,
 
 void BatPublishers::getPublisherBanner(const std::string& publisher_id,
                                        ledger::PublisherBannerCallback callback) {
+
+  BLOG(ledger_, ledger::LogLevel::LOG_ERROR) << "NEJC getPublisherBanner";
```
Comment (ryanml) [inline vendor/bat-native-ledger/src/bat_publishers.cc]: Looks like logs were left in on this file

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat_publishers.cc]: removed

File: vendor/bat-native-ledger/src/bat/ledger/ledger.cc [hunk 2]
```diff
@@ -237,6 +237,7 @@ bool PublisherBanner::loadFromJson(const std::string& json) {
     description = d["description"].GetString();
     background = d["background"].GetString();
     logo = d["logo"].GetString();
+    provider = d["provider"].GetString();
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/ledger.cc]: `HasMember` will prevent banner to execute for old profiles. I can add a check, but needs to be separated

[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/ledger.cc]: done

**Pull request merged**